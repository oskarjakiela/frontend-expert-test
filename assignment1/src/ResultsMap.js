import { join, map, values } from 'ramda';
import React from 'react';
import { GoogleMap, Marker, withGoogleMap, withScriptjs } from 'react-google-maps';
import { compose, withProps } from 'recompose'

import './ResultsMap.css';


const key = compose(
  join(','),
  values,
);

const googleMapElement = () => (<div className="ResultMap__GoogleMap__element" />);

const ResultsMap = compose(
  withProps({
    googleMapURL: 'https://maps.googleapis.com/maps/api/js?v=3.exp',
    loadingElement: googleMapElement(),
    containerElement: googleMapElement(),
    mapElement: googleMapElement(),
  }),
  withScriptjs,
  withGoogleMap,
)(({ markers }) => (
  <div className="ResultsMap">
      <GoogleMap
        defaultZoom={8}
        defaultCenter={{
          lat: 52.33,
          lng: 4.88,
        }}
      >
        {map(position => (<Marker key={key(position)} position={position} />), markers)}
      </GoogleMap>
  </div>
));

export default ResultsMap;
