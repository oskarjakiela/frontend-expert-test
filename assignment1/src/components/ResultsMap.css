.ResultsMap {
}

.ResultMap__GoogleMap__element {
  height: 100%;
}

@media all and (max-width: 736px) {
  .ResultMap__GoogleMap__element {
    height: 100vw;
    max-height: 50vh;
  }
}
