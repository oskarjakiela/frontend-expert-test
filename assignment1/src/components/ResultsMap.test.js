import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import ResultsMap from './ResultsMap';


describe('<ResultsMap />', () => {
  let component;

  beforeEach(() => {
    component = (<ResultsMap
      markers={[
        { lat: 0, lng: 0 },
      ]}
    />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');

    render(component, div);
    unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
