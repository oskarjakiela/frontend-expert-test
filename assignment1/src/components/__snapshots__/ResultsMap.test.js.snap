// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsMap /> renders correctly 1`] = `
<div
  className="ResultMap__GoogleMap__element"
/>
`;
